### PR TITLE
Added a `where` method on the route class which can constraint a route its variables

### PR DESCRIPTION
**Pull request by DJWassink**
Still a WIP, maybe needs some discussion, but most certainly needs some readme ❤️

### CODE REVIEW

File: src/connect.ts [hunk 1]
```diff
@@ -27,6 +28,19 @@ export default function connect(app: Application,
...
 
+        /* Add where constraints */
+        const constraints = route.getConstraints();
+        if (constraints !== undefined) {
+            for (let key in constraints) {
+                // Tell istanbul to ignore the else branch, else the code coverage will go below 100%
+                // https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md#ignore-an-else-path
+                /* istanbul ignore else  */
+                if (constraints.hasOwnProperty(key)) {
+                    path = path.replace(key, key + constraints[key]);
```
Comment (JBlaak) [inline src/connect.ts]: Maybe a bit too naive: `route.get('/user/:identifier/comment/:id').where({id:'[0-9]+')`

**Comment by DJWassink** [comment]
Most of the logic is now in https://github.com/JBlaak/Express-Torch/pull/24/files#diff-8265994ca17d8ef716f946876b8469ed

Also updated path-to-regex to version ^2.2.0, thus also removing the @types/path-to-regexp. Sadly updating path-to-regex to version 2.2.0 resulted in a yarn install conflict with @types/node (peer dep of path-to-regex maybe?), anyhow I also removed @types/node

**Comment by JBlaak** [comment]
:heart_eyes: loving it, add description to the readme and we can merge it!

File: tests/torch.test.ts [hunk 2]
```diff
@@ -613,6 +614,180 @@ describe('Torch', function () {
...
+            };
+
+            /* When */
+            const method = (req: any, res: any) => {
+                /* The impl */
+            };
+            Torch(app as Application, (router: Router) => {
+                router.delete('/:identifier/:id', method).where({
+                    'id': '(\\d+)',
+                    'identifier': '(\\d+)'
```
Comment (JBlaak) [inline tests/torch.test.ts]: Erm, this doesn't test anything, we still do not known which one is inserted where

**Pull request merged**